src/MetadataUtility/Audio/Wamd.cs

```diff
@@ -0,0 +1,203 @@
+
+            var version = GetVersion(wamdSpan);
+
+            if (version.IsFail)
+            {
+                return (Error)version;
+            }
+
+            return version == 1;
+        }
+
+        /// <summary>
+        /// Scans for the position of the wamd chunk in the file stream.
+        /// </summary>
+        /// <param name="stream">The file stream.</param>
+        /// <returns>Range representing the position of the wamd chunk.</returns>
+        public static Fin<RangeHelper.Range> GetWamdChunk(Stream stream)
+        {
+            var riffChunk = Wave.FindRiffChunk(stream);
+            var waveChunk = riffChunk.Bind(r => Wave.FindWaveChunk(stream, r));
+            var wamdChunk = waveChunk.Bind(w => Wave.ScanForChunk(stream, w, WamdChunkId));
+
+            if (wamdChunk.IsFail)
+            {
+                return (Error)wamdChunk;
+            }
+
+            return wamdChunk;
+        }
+
+        /// <summary>
+        /// Extracts the wamd chunk version.
+        /// Wamd version has a sub chunk ID of 0, and is always first in the wamd chunk.
+        /// </summary>
+        /// <param name="wamdSpan">The wamd chunk.</param>
+        /// <returns>Wamd chunk version.</returns>
+        public static Fin<ushort> GetVersion(ReadOnlySpan<byte> wamdSpan)
+        {
+            int wamdOffset = 0;
+
+            ushort subChunkId = BinaryPrimitives.ReadUInt16LittleEndian(wamdSpan[wamdOffset..]);
+            wamdOffset += 2;
+
+            uint length = BinaryPrimitives.ReadUInt32LittleEndian(wamdSpan[wamdOffset..]);
+            wamdOffset += 4;
+
+            if (subChunkId != 0 || length != 2)
+            {
+                return WamdVersionError;
+            }
+
+            ushort version = BinaryPrimitives.ReadUInt16LittleEndian(wamdSpan[wamdOffset..]);
+
+            return version;
+        }
+
+        /// <summary>
+        /// Parses a date into Nodatime's OffsetDateTime.
+        /// </summary>
+        /// <param name="value">The date to parse.</param>
+        /// <returns>The parsed date.</returns>
+        public static OffsetDateTime? DateParser(string value)
+        {
+            OffsetDateTime? date = null;
+
+            try
+            {
+                date = OffsetDateTimePattern.CreateWithInvariantCulture("yyyy'-'MM'-'dd' 'HH':'mm':'sso<m>").Parse(value).Value;
+            }
+            catch (UnparsableValueException)
+            {
+                return null;
+            }
+
+            return date;
+        }
+
+        /// <summary>
+        /// Parses a location's longitude and latitude.
+        /// Assigns the location data to a given wamd object.
+        /// </summary>
+        /// <param name="value">The location to parse.</param>
+        /// <param name="wamdData">The wamd object.</param>
+        public static void SetLocation(string value, Wamd wamdData)
+        {
+            string[] locationInfo = value.Split(",", StringSplitOptions.RemoveEmptyEntries);
```

Comment:
hmm why is empty entry removal needed here?

Reply:
Yeah I'm betting it is because altitude is often empty. We should explicitly attempt to parse altitude in case it is ever not empty. I'm assuming it would be populated when a GPS attachment is added to a SM4 for example.

Reply:
This one's odd, this is how location is presented according to the documentation:

`WGS84,nn.nnnnn,N,mmm.mmmmm,W[,alt]`

I think WGS84 is just the coordinate system being used? In all the examples I've seen that space is empty, meaning location starts with a leading comma, this is what I was filtering out. In case that first space is ever populated with something I'm no longer filtering, just ignoring whatever is in that first space.

I've also added support for altitude.

Reply:
It might be worth noting something about that first space being a coordinate system that is WGS84 and if it is ever not empty we should add code to handle it. This is a good place for a `Debug.assert`, or an exception.